src/sqlfmt/api.py

```diff
@@ -100,18 +100,19 @@ def get_matching_paths(paths: Iterable[Path], mode: Mode) -> Set[Path]:
         for s in mode.exclude:
             if PurePath(s).is_absolute():
                 exclude_set.update([Path(g) for g in glob(s, recursive=True)])
-            elif mode.exclude_root is not None:
+            else:
+                exclude_root = mode.exclude_root or Path.cwd()
                 try:
-                    exclude_set.update(mode.exclude_root.glob(s))
+                    # Prefer glob.glob() to Path.glob() here, even though the
+                    # latter would be more concise, since Path.glob() does not
+                    # resolve symlinks properly in Python <3.13
+                    exclude_set.update(
+                        [Path(g) for g in glob(str(exclude_root / s), recursive=True)]
```

Comment:
It feels a bit hacky to cast the `Path` object to a string here, but it's the only way to get it to work with `glob.glob()`, which expects a string. It's probably fine, and hopefully the tests would tell us if it wasn't, but part of me is a bit nervous about the platform compatibility of this change -- I'm on Ubuntu where it works fine, but I've been trained to be nervous about Windows filesystem compatibility in situations like these.

Reply:
Should be safe to do it this way, but thanks for calling out this change!